lib/Server.js

```diff
@@ -335,16 +343,18 @@ function Server(compiler, options) {
     },
 
     setup: () => {
+      log('Using "setup" is deprecated and will be removed in the next major version. Please use the "before" and "after" hooks instead.');
+      log('If "setup" was working fine for you until now, simply replace it with "before"');
       if (typeof options.setup === 'function') { options.setup(app, this); }
     }
   };
 
-  const defaultFeatures = ['setup', 'headers', 'middleware'];
-  if (options.proxy) { defaultFeatures.push('proxy', 'middleware'); }
+  const defaultFeatures = ['before', 'setup', 'headers', 'middleware', 'after'];
```

Comment:
`after` shouldn't be here. same goes for the changes on lines 353, 357. rather, it should be pushed to `defaultFeatures` before line 365 - just before the `forEach`

Reply:
Fixed.